### PR TITLE
test/unpack_strategy: Handle svn unzip unavailable

### PR DESCRIPTION
**Pull request by sjackman**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Skip those tests that require svn or unzip.

### CODE REVIEW

File: Library/Homebrew/test/unpack_strategy/subversion_spec.rb [hunk 1]
```diff
@@ -15,12 +15,14 @@
     system "svn", "commit", working_copy, "-m", "Add `test` file."
   end
 
-  include_examples "UnpackStrategy::detect"
-  include_examples "#extract", children: ["test"]
+  context "when svn is available", :needs_svn do
```
Comment (reitermarkus) [inline Library/Homebrew/test/unpack_strategy/subversion_spec.rb]: Just add `:needs_svn` to the `describe` above.

**Comment by reitermarkus** [comment]
Maybe change the messages to `"Subversion not installed."` and ``"`unzip` not installed."``.

File: Library/Homebrew/test/unpack_strategy/zip_spec.rb [hunk 2]
```diff
@@ -4,7 +4,10 @@
   let(:path) { TEST_FIXTURE_DIR/"cask/MyFancyApp.zip" }
 
   include_examples "UnpackStrategy::detect"
-  include_examples "#extract", children: ["MyFancyApp"]
+
+  context "when unzip is available", :needs_unzip do
+    include_examples "#extract", children: ["MyFancyApp"]
```
[the remark this answers is not in the document]
Comment (sjackman) [reply Library/Homebrew/test/unpack_strategy/zip_spec.rb]: No

**Comment by sjackman** [comment]
Done.

**Comment by sjackman** [comment]
Merged. Thanks for the review, Markus!

**Pull request merged**